service/auth_client/auth_client.go

```diff
@@ -0,0 +1,86 @@
+package auth_client
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc/credentials/insecure"
+
+	AuthApi "go.buf.build/grpc/go/knox-networks/auth-mgmt/auth_api/v1"
+
+	"google.golang.org/grpc"
+)
+
+type authClient struct {
+	client AuthApi.AuthApiServiceClient
+	conn   *grpc.ClientConn
+}
+
+type AuthClient interface {
+	Close()
+	AuthnWithDid(did string, nonce string, enc []byte) error
+	AuthnWithDidRegister(did string, nonce string, enc []byte) error
+	AuthnWithDidStart() (AuthApi.AuthApiService_AuthnWithDidStartClient, error)
+}
+
+func NewAuthClient(address string) (AuthClient, error) {
```

Comment:
Same as above

Reply:
@sKudryashov The SDK consumer would never use this here. And the point of this package is to abstract all of the gRPC function. And as above we can use structs to handle mocking. So there's no point in doing dependency injection in either case.